contrib/policy_upgrade/generate_hashes.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import json
+import hashlib
+
+from collections import OrderedDict
+
+
+def json_string(data):
+    return json.dumps(data, indent=2)
+
+
+def json_print(data):
+    print(json_string(data))
+
+
+def json_save(data, path):
+    string = json_string(data)
+    with open(path, "w") as f:
+        f.write(string)
+
+
+def json_load(path):
+    with open(path, "r") as f:
+        data = f.read()
+        return json.loads(data, object_pairs_hook=OrderedDict)
+
+
+def file_hash(path):
+    if not os.path.isfile(path):
+        raise ValueError
+
+    hasher = hashlib.sha256()
+    block_size = 4096
+    with open(path, 'rb') as f:
+        buf = f.read(block_size)
+        while len(buf) > 0:
+            hasher.update(buf)
+            buf = f.read(block_size)
+    return hasher.hexdigest()
+
+
+def get_file_paths(root_path):
+    file_paths = []
+    for subdir, dirs, files in os.walk(root_path):
+        for f in files:
+            path = os.path.join(subdir, f)
+            if ".git/" in path:
```

Comment:
This doesn't look right.  How will it handle a directory called `something.git`?